daemons/controld/controld_based.c

```diff
@@ -43,7 +43,13 @@ do_cib_replaced(const char *event, xmlNode * msg)
         return;
     }
 
-    crm_element_value_int(msg, F_CIB_CHANGE_SECTION, &change_section);
+    if ((crm_element_value_ll(msg, F_CIB_CHANGE_SECTION, &change_section) < 0)
+        || (change_section < 0) || (change_section > UINT32_MAX)) {
+
+        crm_debug("Couldn't parse '%s' from message", F_CIB_CHANGE_SECTION);
+        return;
+    }
```

Comment:
For the same reason, we just want to stick with the default here rather than return (crm_element_value_ll() will return -1 if the attribute doesn't exist).

Reply:
Got it, I'll update. For what it's worth, this is *already* broken if `F_CIB_CHANGE_SECTION` doesn't exist. `crm_element_value_int()` stores `-1` in `change_section` if not found.

Reply:
I guess it's never come up because `-1` in binary is all ones, so the `if` condition passes in the next block.

Reply:
:-/

Reply:
Actually nope ... crm_element_value_ll() leaves the output argument unchanged if the attribute is not present (it doesn't call scan_ll if value is NULL)

Reply:
Ah, you're right, good :)